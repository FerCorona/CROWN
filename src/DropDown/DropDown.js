import React, { Component } from 'react';

class DropDown extends Component {
  constructor() {
    super();
    this.state = {
      isActive: false
    };
    this._toggleActive = this._toggleActive.bind(this);
  }

  _toggleActive() {
    this.setState({ isActive: !this.state.isActive });
  }

  render() {
    const { optionSelected, options, onChange, stateName } = this.props;
    return (
      <div className='CROWNDropDown'>
        <div className='Options' onClick={() => this._toggleActive()}>
          {optionSelected}
        </div>
        {
          this.state.isActive && (
            <div className='ListContent'>
              {
                options.map((option, index) => (
                  <div
                    className='Option'
                    key={`option-${index}`}
                    onClick={() => this.setState({ isActive: false },
                      onChange({
                        stateName,
                        value: option
                      }))}
                  >
                    {option}
                  </div>
                ))
              }
            </div>
          )
        }
      </div>
    );
  }
}

DropDown.defaultProps = {
  options: [ 'Button', 'Common', 'DropDown', 'Image', 'InputText', 'Label', 'RadioButton', 'Switch' ],
  placeholder: 'Selecciona una opcion',
  onChange: () => { }
};

export default DropDown;