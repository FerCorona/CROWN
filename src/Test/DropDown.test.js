import React from 'react';
import renderer from 'react-test-renderer';
import DropDown from '../DropDown/index';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <DropDown />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});