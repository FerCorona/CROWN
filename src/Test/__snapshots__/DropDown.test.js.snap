// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="CROWNDropDown "
>
  <div
    className="Options"
    onClick={[Function]}
  />
</div>
`;
